src/main/java/com/conveyal/analysis/controllers/RegionalAnalysisController.java

```diff
@@ -509,16 +521,18 @@ private RegionalAnalysis createRegionalAnalysis (Request req, Response res) thro
         task.cutoffsMinutes = regionalAnalysis.cutoffsMinutes;
         task.percentiles = regionalAnalysis.travelTimePercentiles;
 
-        // Persist this newly created RegionalAnalysis to Mongo, which assigns it an id and creation/update time stamps.
+        // Persist this newly created RegionalAnalysis to Mongo.
+        // This assigns it creation/update time stamps and an ID, which is needed to name any output CSV files.
         regionalAnalysis = Persistence.regionalAnalyses.create(regionalAnalysis);
-        if (analysisRequest.recordTimes) regionalAnalysis.addCsvStoragePath(Result.TIMES, config.resultsBucket());
-        if (analysisRequest.recordPaths) regionalAnalysis.addCsvStoragePath(Result.PATHS, config.resultsBucket());
-        if (analysisRequest.recordAccessibility) regionalAnalysis.addCsvStoragePath(Result.ACCESS, config.resultsBucket());
-        Persistence.regionalAnalyses.modifiyWithoutUpdatingLock(regionalAnalysis);
 
         // Register the regional job with the broker, which will distribute individual tasks to workers and track progress.
         broker.enqueueTasksForRegionalJob(regionalAnalysis);
 
+        // Flush to the database any information added to the RegionalAnalysis object when it was enqueued.
+        // This includes the paths of any CSV files that will be produced by this analysis.
+        // TODO verify whether there is a reason to use regionalAnalyses.modifyWithoutUpdatingLock() or put().
```

Comment:
`put` updates the `nonce`, `createdAt`, and `updatedAt` and should be used by in simple updates from HTTP handlers. `modifyWithoutUpdatingLock` should be used when updating a model multiple times by the backend. There could be a better name for it, suggestions welcome.

Reply:
There's a case that could be made to always use `put` also 🤷

Reply:
The main thing I was wondering is why you'd avoid updating the nonce and update time when something was changed by the backend. I just wasn't sure of the reasoning - we should probably explain it on the method's Javadoc.

Reply:
It may be a premature optimization, mainly skipping updates to fields that aren't needed when the multiple writes are done by the backend without user input.

Looking at all the instances it's used, I'm not sure it's necessary anywhere except in `BundleController.setServiceBundleDates`.

Reply:
Even there, it could probably be removed